### PR TITLE
feat(dedicated.carbon): add tracking for carbon footprint page

### PR DESCRIPTION
**Pull request by darsene**
ref: MANAGER-11291

| Question         | Answer
| ---------------- | ---
| Branch?          | `feat/carbon-calculator`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Fix #MANAGER-11291
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- ~Only FR translations have been updated~
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- ~Breaking change is mentioned in relevant commits~

### CODE REVIEW

File: packages/manager/modules/carbon-calculator/src/dashboard/routing.js [hunk 1]
```diff
@@ -8,6 +10,12 @@ export default /* @ngInject */ ($stateProvider) => {
       dashboardLink: /* @ngInject */ ($state) => $state.href('app.dashboard'),
       breadcrumb: /* @ngInject */ ($translate) =>
         $translate.instant('carbon_calculator_my_carbon_footprint'),
+      trackClick: /* @ngInject */ (atInternet) => {
```
Comment (qpavy) [inline packages/manager/modules/carbon-calculator/src/dashboard/routing.js:13]: Are you using this in controller or html ?

Comment (darsene) [reply packages/manager/modules/carbon-calculator/src/dashboard/routing.js:13]: none of them. Actually, tracking data are sent when page is displayed. Here tracking is done just before displaying the page.
I can move this part in the `controller.$onInit` if you prefer  ?

File: packages/manager/modules/carbon-calculator/src/dashboard/routing.js [hunk 2]
```diff
@@ -8,6 +10,12 @@ export default /* @ngInject */ ($stateProvider) => {
...
+        atInternet.trackClick({
+          name: TRACKING_NAME,
+          type: 'action',
+        });
+      },
```
Comment (anooparveti) [inline packages/manager/modules/carbon-calculator/src/dashboard/routing.js:18]: **nitpick:** If you want to track page, `trackClick` is not the right choice. I think you should use `trackPage` or define `atInternet: { rename: '...'}` on your state definition.

**Review by anooparveti — CHANGES_REQUESTED** [review]

File: packages/manager/modules/carbon-calculator/src/dashboard/carbon-footprint/controller.js [hunk 3]
```diff
@@ -1,14 +1,23 @@
...
     this.$state = $state;
+    this.atInternet = atInternet;
     this.carbonFootprintService = carbonFootprintService;
   }
 
   computeBilling() {
+    this.atInternet.trackClick({
+      name: `${TRACKING_NAME}::download-carbon-consumption`,
+      type: 'action',
+    });
```
Comment (ghyenne) [inline packages/manager/modules/carbon-calculator/src/dashboard/carbon-footprint/controller.js:20]: Maybe you can reuse this.trackClick() created in routing.js you need put in component trackClick: '<',

File: packages/manager/modules/carbon-calculator/src/dashboard/routing.js (hunk 2 above)
Comment (darsene) [reply packages/manager/modules/carbon-calculator/src/dashboard/routing.js:18]: Second option is better, I will do it

[the remark this answers is not in the document]
Comment (darsene) [reply packages/manager/modules/carbon-calculator/src/dashboard/routing.js]: Due to the previous comment, I used `atIntenet: {rename:"XXX"}` instead

File: packages/manager/modules/carbon-calculator/src/dashboard/carbon-footprint/controller.js (hunk 3 above)
Comment (darsene) [reply packages/manager/modules/carbon-calculator/src/dashboard/carbon-footprint/controller.js:20]: I removed the tracking service  from the routing file finally

**Pull request merged**